src/lib/common/sol-missing.h

```diff
@@ -70,6 +70,39 @@ memmem(const void *haystack, size_t haystacklen, const void *needle, size_t need
 }
 #endif
 
+#if !defined(HAVE_GET_CURRENT_DIR_NAME) && defined(HAVE_UNIX)
+#include <errno.h>
+#include <limits.h>         /* for PATH_MAX */
+#include <stdlib.h>
+#include <unistd.h>
+
+static inline char *get_current_dir_name(void)
+{
+    size_t len = 0;
+    char *buffer = NULL;
+
+    while (1) {
+        len += PATH_MAX;
+        char *nb = (char *)realloc(buffer, len);
```

Comment:
you can keep the `struct sol_buffer` on the stack and use `sol_buffer_steal()` :-)

realloc is a PITA